processing/src/main/java/org/apache/druid/query/groupby/DeferExpressionDimensions.java

```diff
@@ -71,7 +71,7 @@ public boolean useDeferredGroupBySelector(
           return false;
         }
 
-        if (!capabilities.isNumeric() && !capabilities.isDictionaryEncoded().isTrue()) {
+        if (!capabilities.isNumeric() && !DeferExpressionDimensions.isDictionaryEncodedScalarString(capabilities)) {
```

Comment:
I don't see why `DeferExpressionDimensions.` is needed?

Reply:
oh i guess it isn't, for some reason intellij did that for code-completion so i just went with it, will fix